docs/example-manifests/SLEMicro/dhcp/rke2-agent.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: sample-cluster
+  name: sample-cluster
+  namespace: default
+spec:
+  clusterName: sample-cluster
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: sample-cluster
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: sample-cluster
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+          kind: RKE2ConfigTemplate
+          name: sample-cluster-workers
+      clusterName: sample-cluster
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: Metal3MachineTemplate
+        name: sample-cluster-workers
+      nodeDrainTimeout: 0s
+      version: v1.25.3+rke2r1
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+kind: RKE2ConfigTemplate
+metadata:
+  name: sample-cluster-workers
+  namespace: default
+spec:
+  template:
+    spec:
+      agentConfig:
+        version: v1.25.3+rke2r1
+        kubelet:
+          extraArgs:
+            - provider-id=metal3://{{ ds.meta_data.uuid }}
+        nodeName: '{{ ds.meta_data.local_hostname }}'
```

Comment:
@alknopfler do you happen to know where these are templated in the SLEMicro case?  I thought these were templated by cloud-init, and @e-minguez and I were discussing if/how it works with SLEMicro

Reply:
I didn't notice any failure using that as is, because I guess the template is added as a prefix to the default value:
https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/e05b03eb7a3cc664ae2f75a11741d7b7cb0a3d53/bootstrap/api/v1alpha1/rke2config_types.go#L84

Reply:
both are optional:
https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/e05b03eb7a3cc664ae2f75a11741d7b7cb0a3d53/bootstrap/api/v1alpha1/rke2config_types.go#L388

TBH I don't know if we should remove both or just keep it with some values as an example to be modified.

Reply:
I'm gonna add it to default values as example.